### ISSUE TITLE
kubefwd integration?

### ISSUE THREAD

**Issue by mattmast** [windmilleng/tilt#929]
Hey, just found this project, it looks really neat! 

Any thoughts on kubefwd integration? Is there one? Does that even make sense? Would there be a workflow that would make sense today?

**Comment by nicks** [comment]
Thanks!

What sort of integration do you have in mind? I can imagine a few different ones that might make sense, depending on how you're using it.

My (vague) understanding of kubefwd is that it does a bunch of bulk network admin operations (forwarding ports, modifying /etc/hosts) so that the local network environment looks kinda like the network inside kubernetes. The idea is that this is good enough that processes outside of kubernetes can kinda pretend they're inside of it (talking to in-cluster services, etc)

Does that sound like what you're doing?

Tilt's intended workflow is that all the servers run inside of kubernetes. Tilt watches your filesystem for edits, then auto-updates the servers. You don't typically modify the local network env at all, except to port-forward servers that you want to hit manually with your browser.

**Comment by mattmast** [comment]
Your understanding is the same as mine.  I mostly use it for convenience for developing integration tests.  

My workflow right now is:

* start tilt
* open another shell for kubefwd
* run tests

At least for now all that is in my Tiltfile is:
```
k8s_yaml(kustomize("directory"))
```

Maybe an ideal looks like:
```
k8s_yaml(kustomize("directory"))
local(kubefwd(namespace='default'))
local("cargo test")
```

**Comment by nicks** [comment]
ah, i see! You're running the integration tests outside Kubernetes, and kubefwd lets the tests talk to all the services?

Out of curiosity, have you ever considered running `cargo test` inside the cluster?

One detail that I'm not totally sure how we'd handle is giving kubefwd `sudo` privileges, because I think it needs them?

**Comment by landism** [comment]
> how we'd handle is giving kubefwd sudo privileges, because I think it needs them?

Yeah, it needs sudo privileges because it generates hostnames for services by editing /etc/hosts.

I'm guessing we don't want to tell users to run tilt via sudo. To avoid having to make Tilt interactive for this, perhaps users would be comfortable adding `$USER ALL = NOPASSWD: kubefwd` to their /etc/sudoers.
